spec/spec_helper.rb

```diff
@@ -2,6 +2,13 @@
 
 require "escalate"
 
+RSpec::Matchers.define :be_exception do |klass, message|
+  match do |actual|
+    expect(actual).to be_a(klass)
+    expect(actual.message).to eq(message)
+  end
+end
```

Comment:
I've repeatedly been stymied that you can't just match an exception in an rspec `with()` list, as in
```
expect(SomeGem).to receive(:escalate).with(ArgumentError.new('bang!'))
```
I think the reason is that the backtrace from a `raise`d exception won't match the empty one from `new`.

So I took a shot at writing a custom matcher. That was easy! This `be_exception` matcher knows to to match on just the class and message. This might be useful elsewhere.

Reply:
This is awesome! Yea I think this would be a valuable matcher everywhere that we use `rspec`. It's great how easy they are to create too.

Reply:
We don't have a common place to put these yet, do we? A library of our own matchers could be good... but also could become a dumping ground if were weren't careful. I've been tempted to have our common `rspec` configuration live in a common place, so maybe we could do that at the same time?

For example, I recently added the `2_000` setting here because the default tends to truncate diffs right when you need them! https://github.com/Invoca/escalate/blob/main/spec/spec_helper.rb

Reply:
We don't have that yet, but I like the idea of consolidating it all in the same spot. Would safe a lot of copy/paste time when setting up new gems.